HomeWork3/Alex_Zubok_home_work3_1.cpp

```diff
@@ -0,0 +1,35 @@
+#include <cstring>
+#include <iostream>
+
+void reverse_string(char *string) {
+  if (string == nullptr || string + std::strlen(string) == nullptr || std::strlen(string) == 0)
+    return;
+  char *tmp_string = new char[std::strlen(string)];
+  std::strncpy(tmp_string, string, std::strlen(string));
+  std::cout << "Your reverse string is : ";
+  for (size_t index = std::strlen(tmp_string) - 1, i = 0; i < std::strlen(tmp_string); index--, i++) {
+    *(string + i) = *(tmp_string + index);
+    std::cout << *(tmp_string + index);
+  }
+  *(string + (std::strlen(tmp_string) - 1)) = *tmp_string;
+  std::cout << std::endl;
+  delete[] tmp_string;
+}
+
+int main() {
+  char *tmp_string = new char;
+  std::cout << "Please enter your string that you want reverse: ";
+  std::cin >> tmp_string;
+
+  char *string_full = new char[std::strlen(tmp_string) + 1];
```

Comment:
```suggestion
  char *string_full = new char[std::strlen(tmp_string) + 1]{};
```
чтобы у вас небыло мусора в массиве

Reply:
да, спасибо, забыл про это и столкнулся с такой вроблемой в 4-м задании, долго не мог понять, в чем проблема